posttroll/listener.py

```diff
@@ -103,26 +98,32 @@ def __init__(self, topics=None, queue=None, addresses=None,
         self.running = False
 
     def create_subscriber(self):
-        '''Create a subscriber instance using specified addresses and
-        message types.
-        '''
+        """Create a subscriber instance using specified addresses and message types."""
         if self.subscriber is None:
             if self.topics:
-                self.subscriber = NSSubscriber(self.services, self.topics,
-                                               addr_listener=True,
-                                               addresses=self.addresses,
-                                               nameserver=self.nameserver)
-                self.recv = self.subscriber.start().recv
+                config = self._get_subscriber_config()
+                self.subscriber = subscriber.dict_config(config)
+                try:
+                    self.recv = self.subscriber.start().recv
+                except AttributeError:
+                    self.recv = self.subscriber.recv
```

Comment:
Should `dict_config` return a started `NSSubscriber` instance, so that this `try`/`except` isn't needed?

Reply:
Sure, I'll change that.

Reply:
Ok, can't do that because the `Subscribe` context manager won't work then. And evidently the context manager won't work with `Subscriber` as it doesn't have a `start()` method (and there's no test for it)...

Reply:
Could we add a test for that?

Reply:
Yep, that's what I was about to start now after lunch.

Reply:
Done.